iot_api/user_api/Error.py

```diff
@@ -35,6 +63,14 @@ class Internal(Exception):
 # Error handlers: these functions are called when an exception is raised.
 # In most cases they respond with a short message and the corresponding
 # HTML error code.
+
+@app.errorhandler(InvalidUsage)
+def handle_invalid_usage(error):
+    log.error(str(error))
+    response = jsonify(error.to_dict())
```

Comment:
I think you don't need to jsonify the response, you can return a dict and flask handle that conversion.